### PR TITLE
Final

### PR DESCRIPTION
**Pull request by AckonSamuel**
### READY FOR REVIEW

<hr>

In this PR, these are the requirement implemented:

## General requirements

- [x]  Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).

- [x]  Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).

- [x]  Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### HTML/CSS & JavaScript requirements

- [x]  Follow our list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).

- [x]  Follow our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements

**API**
- [https://ghibliapi.herokuapp.com/](API)

**Design**

- [x] You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:

  - Colors (select one main color for your website).
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

**Interactions**

- [x] Home page

  - When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
  - Along with the category name, you will display some numeric values per category that come from the API.
  - When the user clicks (or taps) on a category item, the application navigates to the details page.

- [x]  Details page

  - In the details page, the webapp retrieves data from the API to show detailed data in the category.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

**Testing requirements**

- [x] Create unit tests for pure functions in your Redux code.

- [x]  Create integration tests for your application using the React Testing Library.

**Technical requirements**

- [x] The project is a single-page application (SPA) built with React and Redux.

- [x] The data retrieved from the API should be stored in the Redux store.

- [x] You should filter the data that you retrieve from the API using a Filter stateless component.

- [x] Every page (the main page and the pages for each item) should have a unique route within the SPA.

- [x] The project should be deployed and accessible online (add a link to the online version of your app to your README file).

### Project documentation

Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a **maximum length of 5 minutes**. The content of the video should include:

- [x]  a description of the project.

- [x]  a demo of the project features.

- [x]  you should also highlight some interesting piece of code or something you built that you are very proud of.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,62 @@
...
 
-### `npm run eject`
+#### install dependencies
+``` run 
+$ npm i
+```
+#### run project
+```
+$ npm start
+```
```
Comment (Bondok6) [inline README.md:44]: - [x] I suggest adding one more command to run tests `npm test` 👌
- [x] Your README should contain an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/), kindly add it 👍

File: README.md [hunk 2]
```diff
@@ -1,46 +1,62 @@
...
+Contributions, issues, and feature requests are welcome!
 
-## Learn More
+Feel free to check the [issues page](../../issues/).
 
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
+## Show your support
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+Give a ⭐️ if you like this project!
```
Comment (Bondok6) [inline README.md:66]: - [x] I suggest adding the link to your `MIT.md` file to your README 👌, and remember to add your name and customize the MIT.md  😉

image


> The MIT license gives users express permission to reuse code for any purpose, sometimes even if the code is part of proprietary software. As long as users include the original copy of the MIT license in their distribution, they can make any changes or modifications to the code to suit their own needs. 💯

**Review by Bondok6 — CHANGES_REQUESTED** [review]
## Hi @AckonSamuel 🙋‍♂️, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!



## To Highlight
- [x] Good video presentation✅
- [x] Deployment correctly ✅
- [x] Well file structure ✅
- [x] Great job using react-router-dom and redux ✅
- [x] No Linter errors ✅
- [x] Correct GitHub Flow ✅

## Suggested changes
- [x] According to the requirements You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
Colors (select one main color for your website).
Typography: font face, size, and weight.
Layout: composition and space between the element
> the layout for the home page should have two columns

| Your Design  | Figma Design  |   
| --- | --- |
|   |  |

- [x] I suggest adding more unit tests for several components of your app 👌

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 2 above)
Comment (AckonSamuel) [reply README.md:66]: Got it!

File: README.md (hunk 1 above)
Comment (AckonSamuel) [reply README.md:44]: Sure brother!

File: README.md (hunk 2 above)
Comment (AckonSamuel) [reply README.md:66]: Thanks for the review

File: src/components/Navbar.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import PropTypes from 'prop-types';
+import Container from 'react-bootstrap/Container';
+import Form from 'react-bootstrap/Form';
+import Navbar from 'react-bootstrap/Navbar';
+
+const NavBar = ({ textListener, searchText }) => (
+  <Navbar bg="dark" expand="lg" style={{ width: '100%', position: 'fixed', zIndex: '10000' }}>
+    <Container fluid style={{ color: 'black' }}>
+      <Navbar.Brand href="#home">Watch-hub</Navbar.Brand>
```
Comment (danmainah) [inline src/components/Navbar.js:9]: - [ ] [optional] I would suggest you change the text color of your navbar text, currently, it's difficult to figure out text in your app.

**Review by danmainah — APPROVED** [review]
Hi @AckonSamuel , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**